netlayers/tcp.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Jessica Tallon
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import socket
+from urllib.parse import urlparse
+
+from contrib import syrup
+from netlayers.base import CapTPSocket, Netlayer
+
+from utils.ocapn_uris import OCapNNode
+from utils.captp import CapTPSession
+
+class TCPNetlayer(Netlayer):
+
+    def __init__(self, 
+        listen_address="127.0.0.1", 
+        listen_port=22045,
+        listen_queue_size=100
+    ):
+        self.server_sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.server_sock.bind((listen_address, listen_port))
+        self.server_sock.listen(listen_queue_size)
+        
+        # refreshing both
+        (listen_address, listen_port) = self.server_sock.getsockname()
+        self._connections = []
+      
+        self.address, self.port = listen_address, listen_port
+        self.location = OCapNNode(syrup.Symbol("tcp"), f"{listen_address}:{listen_port}", False)
+
+    def __del__(self):
+        self.shutdown()
+
+    def connect(self, ocapn_machine: OCapNNode) -> CapTPSession:
+        """ Connect to the remote machine """
+
+        url = urlparse(f"tcp://{ocapn_machine.address}")
```

Comment:
We should probably change the transport name to something which makes it clear this is for testing, something like "tcp-testing-only://"